overlord/hookstate/ctlcmd/get.go

```diff
@@ -22,23 +22,32 @@ package ctlcmd
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/snapcore/snapd/i18n/dumb"
 	"github.com/snapcore/snapd/overlord/configstate"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
 )
 
 type getCommand struct {
 	baseCommand
 
+	// these two options are mutually exclusive
+	ForceSlotSide bool `long:"slot" description:"request attribute of the slot"`
+	ForcePlugSide bool `long:"plug" description:"request attribute of the plug"`
+
 	Positional struct {
-		Keys []string `positional-arg-name:"<keys>" description:"option keys" required:"yes"`
-	} `positional-args:"yes" required:"yes"`
+		PlugOrSlotSpec string   `positional-args:"true" positional-arg-name:":<plug|slot>" required:"yes"`
+		Keys           []string `positional-arg-name:"<keys>" description:"option keys"`
+	} `positional-args:"yes"`
 
 	Document bool `short:"d" description:"always return document, even with single key"`
 	Typed    bool `short:"t" description:"strict typing with nulls and quoted strings"`
 }
 
-var shortGetHelp = i18n.G("Prints configuration options")
+var shortGetHelp = i18n.G("Prints configuration options or attribute values of plug/slot")
```

Comment:
Let's try to merge the two cases in a single go:

`"Prints configuration and interface connection settings."`